lib/cadet/assessments/assessments.ex

```
@@ -371,6 +372,78 @@ defmodule Cadet.Assessments do
     end
   end
 
+  def unsubmit_submission(submission_id, avenger = %User{role: role, id: user_id})
+      when is_ecto_id(submission_id) do
+    if role in @unsubmit_assessment_role do
+      submission =
+        Submission
+        |> where(id: ^submission_id)
+        |> join(:inner, [s], a in assoc(s, :assessment))
+        |> preload([_, a], assessment: a)
+        |> Repo.one()
+
+      with {:submission_found?, true} <- {:submission_found?, is_map(submission)},
+           {:is_open?, true} <- is_open?(submission.assessment),
+           {:status, :submitted} <- {:status, submission.status},
+           {:avenger_of?, true} <- {:avenger_of?, is_avenger_of(avenger, submission.student_id)} do
+        rollback_submission_status(submission)
+        rollback_grading_answers(submission)
+
+        {:ok, nil}
+      else
+        {:submission_found?, false} ->
+          {:error, {:not_found, "Submission not found"}}
+
+        {:is_open?, false} ->
+          {:error, {:forbidden, "Assessment not open"}}
+
+        {:status, :attempting} ->
+          {:error, {:bad_request, "Some questions have not been attempted"}}
+
+        {:status, :attempted} ->
+          {:error, {:forbidden, "Assessment has not been submitted"}}
+
+        {:avenger_of, false} ->
+          {:error, {:forbidden, "Only avenger of student is allowed to unsubmit"}}
+
+        _ ->
```

Just let it crash. I don't think this clause is necessary